From 10d55747d353c9b2db4df46ded931718773a059f Mon Sep 17 00:00:00 2001
From: Naadir Jeewa <blah>
Date: Fri, 25 Jun 2021 10:40:12 +0100
Subject: [PATCH] This is a hack

---
 extension.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extension.js b/extension.js
index ceda859..90da7bb 100644
--- a/extension.js
+++ b/extension.js
@@ -194,7 +194,7 @@ function notify(msg, details, params) {
     Main.messageTray.add(source);
     let notification = new MessageTray.Notification(source, msg, details, params);
     notification.setResident(true); // Usually more annoying that the notification disappear than not
-    source.notify(notification);
+    source.notify(msg);
     return notification;
 }
 
-- 
2.31.1
