diff --git a/meson_options.txt b/meson_options.txt
index 2c2a4683..d499332c 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -159,3 +159,6 @@ option('sndfile',
        description: 'Enable code that depends on libsndfile',
        type: 'feature',
        value: 'auto')
+option('pipewire_pulse_prefix',
+       type : 'string',
+       description : 'Install directory for the pipewire-pulse daemon')
diff --git a/src/daemon/systemd/user/meson.build b/src/daemon/systemd/user/meson.build
index 46dfbbc8..0d975cec 100644
--- a/src/daemon/systemd/user/meson.build
+++ b/src/daemon/systemd/user/meson.build
@@ -9,7 +9,7 @@ install_data(
 
 systemd_config = configuration_data()
 systemd_config.set('PW_BINARY', join_paths(pipewire_bindir, 'pipewire'))
-systemd_config.set('PW_PULSE_BINARY', join_paths(pipewire_bindir, 'pipewire-pulse'))
+systemd_config.set('PW_PULSE_BINARY', join_paths(get_option('pipewire_pulse_prefix'), 'bin/pipewire-pulse'))
 
 configure_file(input : 'pipewire.service.in',
                output : 'pipewire.service',
