diff --git a/src/gui/component.rs b/src/gui/component.rs
index 692309c..f2fb9c0 100644
--- a/src/gui/component.rs
+++ b/src/gui/component.rs
@@ -371,7 +371,7 @@ impl AsyncComponent for Greeter {
         if let Some(monitor) = &model.updates.monitor {
             // The window needs to be manually fullscreened, since the monitor is `None` at widget
             // init.
-            root.fullscreen_on_monitor(monitor);
+            // root.fullscreen_on_monitor(monitor);
         } else {
             // Couldn't choose a monitor, so let the compositor choose it for us.
             root.fullscreen();
